### ISSUE TITLE
Problem to get element with className

### ISSUE THREAD

**Issue by rp-happiestbaby** [kata198/AdvancedHTMLParser#6]
Hi,
Im having problem when i'm trying to get element with classname when there is a "-" or space in class name.

see code sample

`
#!/usr/bin/python
  
import AdvancedHTMLParser
import requests
r = requests.get('SOMEURL')
parser = AdvancedHTMLParser.AdvancedHTMLParser()
parser.parseStr(r.text)
e1=parser.getElementsByClassName('body-content')

print (e1[0].innerHTML)
`

**Comment by kata198** [comment]
I'm not able to reproduce this issue. Are you using the latest version of AdvancedHTMLParser?

Can you please provide a testcase? The following is passing for me:

```
class TestAttributes(object):

    ...
            html = '''<html><head><title>Page</title></head>
<body class="background">
  <div id="outer" class="outer">
   <div class="inner special">Hello</div>
   <div class="inner cheese">
     <div class="blah" id="blahdiv1">One</div>
       <span>
         <div class="blah" id="blahdiv2" >
         </div>
       </span>
     </div>
   </div>
  </div>
  <div id="outer2" >
    <div class="yes-dash cheese" >Hello</div>
  </div>
  <div id="outer3" >
    <div class="class-dash CDONE" >One</div>
    <div class="CDBEFORE class-dash CDAFTER class-dash2" >Two</div>
    <div class="class-dash">Three
      <div class="class-dash pickle">Nested Four</div>
    </div>
</body>
</html>
        '''
        document = AdvancedHTMLParser()
        document.parseStr(html)

         .,....
        # Test dash in class name - Issue #6
        dashDivs = document.getElementsByClassName('yes-dash')
        assert dashDivs , 'Failed to find any results for getElementsByClassName with dash.'
        assert len(dashDivs) == 1 , 'Expected to find one result for class="yes-dash", but got %d. %s' %(len(dashDivs), repr(dashDivs))

        dashDiv = dashDivs[0]
        assert dashDiv.innerText.strip() == 'Hello' , 'Expected innerText="Hello", but got: %s' %(repr(dashDiv.innerText.strip()), )
        assert dashDiv.tagName == 'div' , 'Expected to find a div, but got %s' %(dashDiv.tagName, )


        classDashEms = document.getElementsByClassName('class-dash')
        assert classDashEms , 'Failed to find elements with class name = "class-dash"'
        assert len(classDashEms) == 4 , 'Expected to find 4 elements with class name = "class-dash" but got %d. %s' %( len(classDashEms), repr(classDashEms))

        classDash2Ems = document.getElementsByClassName('class-dash2')
        assert classDash2Ems , 'Failed to find elements with class name = "class-dash2"'
        assert len(classDash2Ems) == 1 , 'Expected to find 1 element with class name = "class-dash2" but got %d. %s' %( len(classDash2Ems), repr(classDash2Ems))

```

And the test is passing as expected.

Please provide an HTML string or a url which reproduces the issue and the code you are using. Thanks.

**Comment by rp-happiestbaby** [comment]
Hi @kata198,
Thx for responce 
Problem that I'm facing see code bellow (dash looks like working)

```
#!/usr/bin/python
  
import AdvancedHTMLParser
import requests
import os

# First Lets check with bash 
os.system("curl -s https://en.wikipedia.org/wiki/Static_web_page | grep 'noprint stopMobileRedirectToggle'")

# now lets check lib 
r = requests.get('https://en.wikipedia.org/wiki/Static_web_page')
parser = AdvancedHTMLParser.AdvancedHTMLParser()
parser.parseStr(r.text)
e1=parser.getElementsByClassName('noprint stopMobileRedirectToggle')
# lets print first match innerHTML
print (e1[0].innerHTML)
```

**Comment by kata198** [comment]
I see the problem -- getElementsByClassName only works with a single class name, it does not support the extension to where you can use spaces to select multiple class names.

I will fix this in the next version, some time over the next couple of days. As a workaround until then, you can do it through chaining off the TagCollection's initial result, so

e1 = parser.getElementsCustomFilter(lambda em : 'noprint' in em.classList and 'stopMobileRedirectToggle' in em.classList)

This will yield the same result, you can even stuff other conditionals or a full function in there.

I apologize for this not being included in the original implementation, this is the first I am seeing that form used with getElementsByClassName in many years of web development :) Anyway, workaround above, or I'll release 8.2.0 with this fix in place.

Thanks again!

**Comment by kata198** [comment]
This has been resolved in 8.1.7. Sorry it took so long, have a lot going on.
